Algebra.java

```diff
@@ -6,62 +6,188 @@
 public class Algebra {
```

Comment:
This is an Excellent submission! 
Well Done!! 
Please contact me (nothing bad)

Tal Danai